### PR TITLE
RavenDB-7865 - S3 & Glacier backup improvements

### PR DESCRIPTION
**Pull request by grisha-kotler**
- supporting multi part uploads for S3 (up to 5TB) and Glacier (up to 40TB)
- testing for S3 and Glacier
- added progress to the upload process

### CODE REVIEW

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/s3Settings.ts [hunk 1]
```diff
@@ -25,27 +25,39 @@ class s3Settings extends amazonSettings {
...
                 },
                 {
                     validator: (bucketName: string) => this.validate(() =>
                         !!bucketName && !!bucketName && !bucketName.includes("..")),
-                    message: "There can be only one period between labels"
+                    message: "Bucket name cannot contain consecutive periods"
+                },
+                {
+                    validator: (bucketName: string) => this.validate(() =>
+                        !!bucketName && !!bucketName && !bucketName.includes(".-") && !bucketName.includes("-.")),
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/periodicBackup/s3Settings.ts]: !!bucketName && !!bucketName  :)

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/amazonSettings.ts [hunk 2]
```diff
@@ -94,12 +94,6 @@ abstract class amazonSettings extends backupSettings {
                 onlyIf: () => this.enabled()
             }
         });
-
-        this.credentialsValidationGroup = ko.validatedObservable({
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/periodicBackup/amazonSettings.ts:98]: Will we have credentails validation or just test connection?

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editPeriodicBackupTask.html [hunk 3]
```diff
@@ -117,7 +117,7 @@
                     </div>
                     <div class="row flex-row margin-bottom">
                         <label class="control-label col-xs-12 col-sm-6 col-lg-4">Remote folder name</label>
-                        <div class="col-xs-12 col-sm-6 col-lg-8" data-bind="validationElement: remoteFolderName">
+                        <div class="col-xs-12 col-sm-6 col-lg-8">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/editPeriodicBackupTask.html]: Why is it removed?

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editPeriodicBackupTask.html [hunk 4]
```diff
@@ -179,7 +179,7 @@
                     </div>
                     <div class="row flex-row margin-bottom">
                         <label class="control-label col-xs-12 col-sm-6 col-lg-4">Remote folder name</label>
-                        <div class="col-xs-12 col-sm-6 col-lg-8" data-bind="validationElement: remoteFolderName">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/editPeriodicBackupTask.html]: Why is it removed?

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/amazonSettings.ts (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/amazonSettings.ts:98]: this is going to test the credentials and the existence of the bucket/vault in the specified region

**Pull request merged**